exchanges/mock/recording.go

```diff
@@ -299,10 +299,15 @@ const (
 // CheckJSON recursively parses json data to retract keywords, quite intensive.
 func CheckJSON(data interface{}, excluded *Exclusion) (interface{}, error) {
 	var context map[string]interface{}
+
 	if reflect.TypeOf(data).String() == "[]interface {}" {
+		d, ok := data.([]interface{}) // make linters happy, even with the above check
```

Comment:
This could be changed to:
```

// CheckJSON recursively parses json data to retract keywords, quite intensive.
func CheckJSON(data interface{}, excluded *Exclusion) (interface{}, error) {
	d, ok := data.([]interface{})
	if ok {
		var sData []interface{}
		for i := range d {
			v := d[i]
			switch v.(type) {
			case map[string]interface{}, []interface{}:
				checkedData, err := CheckJSON(v, excluded)
				if err != nil {
					return nil, err
				}

				sData = append(sData, checkedData)
			default:
				// Primitive value doesn't need exclusions applied, e.g. float64 or string
				sData = append(sData, v)
			}
		}
		return sData, nil
	}

	conv, err := json.Marshal(data)
	if err != nil {
		return nil, err
	}

	var context map[string]interface{}
	err = json.Unmarshal(conv, &context)
...
```